devices/src/virtio/mmio.rs

```diff
@@ -229,10 +229,10 @@ impl MmioDevice {
                 // check will fail and take the device into an unusable state.
                 if !self.device_activated && self.are_queues_valid() {
                     if let Some(ref interrupt_evt) = self.interrupt_evt {
-                        if let Some(ref mem) = self.mem {
+                        if let Some(mem) = self.mem.take() {
```

Comment:
I am all for removing a all to `clone` if not needed. But just out of curiosity, is there any other benefit of doing this change ? From what I understand GuestMemory only contains an Arc so `clone` should only increment the counter. There is almost no overhead to that.

Reply:
The intended effect here is not overhead reduction, but tightening of ownership.
The virtio-spec states that once the device is activated, there should be no further attempts to configure it - this means our generic (spec) virtio impl only needs access to memory during the configuration phase.

We have a state machine in place that should guard against misbehaving guests trying to break the spec implementation. Relinquishing access to memory is just one extra step we can take to make sure that the config is not changed while in use.